pkg/daemon/ceph/cleanup/radosnamespace.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cleanup
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+)
+
+func RadosNamespaceCleanup(context *clusterd.Context, clusterInfo *client.ClusterInfo, poolName, radosNamespace string) error {
+	logger.Infof("starting clean up of CephBlockPoolRadosNamespace %q resources in blockpool %q", radosNamespace, poolName)
+
+	images, err := cephclient.ListImagesInRadosNamespace(context, clusterInfo, poolName, radosNamespace)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list images in blockpool %q in rados namespace %q", poolName, radosNamespace)
+	}
+
+	var retErr error
+	for _, image := range images {
+		err := cephclient.PurgeImageSnapshots(context, clusterInfo, poolName, image.Name, radosNamespace)
+		if err != nil {
+			retErr = errors.Wrapf(err, "failed to purge snapshots of the image %q in pool %q.", image.Name, poolName)
+			logger.Error(retErr)
+		}
+
+		err = cephclient.DeleteImageInRadosNamespace(context, clusterInfo, image.Name, poolName, radosNamespace)
+		if err != nil {
+			retErr = errors.Wrapf(err, "failed to delete image %q in pool %q.", image.Name, poolName)
+			logger.Error(retErr)
+		}
```

Comment:
AFAIK rhis doesn't work when we have clones created from the snapshot, when we try to delete the clones on the parent image the parent will fail saying there are clones created from the parent image

```
sh-4.4$ rbd clone --rbd-default-clone-format 2 replicapool/test@test replicapool/test2
sh-4.4$ 
sh-4.4$ rbd ls replicapool
test
test2
sh-4.4$ rbd snap purge test
rbd: error opening default pool 'rbd'
Ensure that the default pool has been created or specify an alternate pool name.
sh-4.4$ rbd snap purge replicapool/test
Removing all snapshots: 100% complete...done.
sh-4.4$ rbd rm replicapool/test
2024-04-10T18:04:15.724+0000 7f406b33b580 -1 librbd::api::Image: remove: image has snapshots - not removing
Removing image: 0% complete...failed.
rbd: image has snapshots with linked clones - these must be deleted or flattened before the image can be removed.
sh-4.4$ 
sh-4.4$ ceph rbd task add trash remove replicapool/911d3a445d20
{"sequence": 1, "id": "2fc6afa4-1239-45f9-98e6-2a4298d8839c", "message": "Removing image replicapool/911d3a445d20 from trash", "refs": {"action": "trash remove", "pool_name": "replicapool", "pool_namespace": "", "image_id": "911d3a445d20"}}
sh-4.4$ rbd trash mv replicapool/test2
sh-4.4$ ceph rbd task add trash remove replicapool/ee91d79d6186
{"sequence": 2, "id": "fed70961-a430-4b6d-8594-f8e18f824905", "message": "Removing image replicapool/ee91d79d6186 from trash", "refs": {"action": "trash remove", "pool_name": "replicapool", "pool_namespace": "", "image_id": "ee91d79d6186"}}
sh-4.4$ ceph rbd task add trash remove replicapool/ee91d79d6186
{"sequence": 2, "id": "fed70961-a430-4b6d-8594-f8e18f824905", "message": "Removing image replicapool/ee91d79d6186 from trash", "refs": {"action": "trash remove", "pool_name": "replicapool", "pool_namespace": "", "image_id": "ee91d79d6186"}}
```

please test it out and see all the images are deleted from pool/namespace and also from trash.